crawler.py

```
@@ -232,16 +238,96 @@ def _fetch_page(self, url: str) -> Optional[str]:
+            
+            # リトライ判定
+            if retry_count < max_retries - 1 and self.error_handler.should_retry(error):
+                time.sleep(2 ** retry_count)  # 指数バックオフ
+                return self._fetch_page(url, retry_count + 1)
+            
             return None
+            
         except Exception as e:
-            self.logger.error(f"予期しないエラー: {url} - {e}")
+            error = NetworkError(
+                message=f"予期しないエラー",
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)